index.js

```diff
@@ -17,7 +30,36 @@ function isEmpty(obj){
     return true
 }
 
-function createXHR(options, callback) {
+function initParams(uri, options, callback) {
+    if (isFunction(options)) {
+        callback = options
+    }
+
+    var params
+    if (typeof options === "object") {
```

Comment:
`isFunction(options)` and `typeof options === "object"` can be true at the same time. But no worries, I figured it out.